src/base_client.py

```
@@ -0,0 +1,102 @@
+                method=method.upper(), 
+                url=url, 
+                params=params, 
+                headers=request_headers,
+                timeout=self.timeout
+            )
+            
+            # Raise an exception for HTTP errors
+            response.raise_for_status()
+            
+            # Log successful response
+            self.logger.info(f"Received {response.status_code} response")
```

Pylint (3.3.7)

[warning] Use lazy % formatting in logging functions

(W1203)